$OpenBSD: patch-libzeitgeist_log_vala,v 1.3 2013/09/12 06:49:54 ajacoutot Exp $

get_nprocs_conf() is a glibc-extension equivalent to sysconf(_SC_NPROCESSORS_CONF).
https://bugs.launchpad.net/libzeitgeist/+bug/1161565

--- libzeitgeist/log.vala.orig	Tue Mar 19 07:01:48 2013
+++ libzeitgeist/log.vala	Thu Mar 28 20:03:52 2013
@@ -40,6 +40,10 @@
  * Zeitgeist also comes with a blacklist extension to make sure the user
  * always stays in control of what information is logged.
  */
+
+using Posix;
+using Linux;
+
 namespace Zeitgeist
 {
 
@@ -58,9 +62,6 @@ namespace Zeitgeist
  */
 public class Log : QueuedProxyWrapper
 {
-    [CCode (cheader_filename = "sys/sysinfo.h", cname = "get_nprocs_conf")]
-    extern static int get_nprocs_conf ();
-
     class DbWorker
     {
         private unowned ThreadFunc<void*> func;
@@ -141,7 +142,7 @@ public class Log : QueuedProxyWrapper
         try {
             threads = new ThreadPool<DbWorker>.with_owned_data ((worker) => {
                 worker.run ();
-            }, get_nprocs_conf (), true);
+            }, (int) Posix.sysconf(Linux._SC_NPROCESSORS_CONF), true);
         } catch (ThreadError err) {
             warning ("%s", err.message);
             threads = null;
